src/components/TaskList/index.js

```diff
@@ -50,39 +20,77 @@ function TaskList(sources) {
-  // A STREAM OF ALL ACTIONS ON ALL TASKS
-  // Each todo item has an action stream. All those action streams are being
-  // merged into a stream of all actions. Below this stream is passed into
-  // the proxyItemAction$ that we passed to the intent function above.
-  // This is how the intent on all the todo items flows back through the intent
-  // function of the list and can be handled in the model function of the list.
-  let itemAction$ = amendedState$
-    .map(({list}) => xs.merge(...list.map(i => i.todoItem.action$)))
-    .flatten();
-  // PASS REAL ITEM ACTIONS TO PROXY
-  // The item actions are passed to the proxy object.
-  proxyItemAction$.imitate(itemAction$);
+  // THE ITEM ADDITION STREAM
+  // Emits objects of sources specific to each item.
+  // Merges stored items with new items
+  let add$ = xs.merge(
+    sourceTodosData$
+      .map(data => data.list.map(props => ({props$: xs.of(props)}))),
+    action$
+      .filter(action => action.type === 'insertTodo')
+      .map(action => ({
+        props$: xs.of({
+          title: action.payload,
+          completed: false
+        })
+      }))
+  );
+  // THE ITEM REMOVAL SELECTOR FUNCTION
+  // This function takes item's sinks and returns a stream representing
+  // its removal. Merges internal removals and `deleteCompleteds` actions
+  function removeSelector(itemSinks) {
+    let deleteCompleteds$ = action$
+      .filter(action => action.type === 'deleteCompleteds');
+    return xs.merge(
+      // Consider deleteCompleteds$ only if the task is completed.
+      // analogue of rx pausable
+      itemSinks.state$
+        .map(state => deleteCompleteds$.filter(() => state.completed))
+        .flatten(),
+      itemSinks.action$
+        .filter(action => action.type === 'destroy')
+    );
+  }
+  // THE COLLECTION STREAM
+  // Collection function takes a component function, a common sources object,
+  // a stream of item additions, and a selector function from item sinks to
+  // a stream of removals
+  let list$ = Collection(
+    Task,
+    {
+      DOM: sources.DOM,
+      action$: action$
+        .filter(action => action.type === 'toggleAll')
+    },
+    add$,
+    removeSelector
+  );
+  // THE COMBINED CHILDREN VTREE AND STATE STREAMS
+  let todoVtrees$ = Collection.pluck(list$, itemSinks => itemSinks.DOM);
+  let todoStates$ = Collection.pluck(list$, itemSinks => itemSinks.state$);
+
+  let amendedState$ = xs.combine(state$, todoVtrees$, todoStates$)
+    .map(([parentState, todoVtrees, todoStates]) => ({
+      ...parentState,
+      list: todoStates.map((state, i) => ({
+        ...state,
+        todoItem: {DOM: todoVtrees[i]}
+      }))
+    }));
   // THE VIEW (MVI PATTERN)
   // We render state as markup for the DOM.
   let vdom$ = view(amendedState$);
   // WRITE TO LOCALSTORAGE
   // The latest state is written to localStorage.
-  let storage$ = serialize(state$).map((state) => ({
+  let storage$ = serialize(amendedState$).map((state) => ({
```

Comment:
Am I right in thinking that we don't want to serialize the `DOM`, so this should probably be `todoStates$` instead of `amendedState$`?